### PR TITLE
[FEATURE] Supported authentication protected servers

### PR DESCRIPTION
**Pull request by pgampe**
Allow to access solr servers which require HTTP authentication.

Fixes: #788

### CODE REVIEW

File: Classes/ConnectionManager.php [hunk 1]
```diff
@@ -254,6 +256,7 @@ public function getAllConnections()
                 $solrServer['solrPath'],
                 $solrServer['solrScheme']
             );
+            $this->checkAndSetAuthentication(end($connections), $solrServer);
```
Comment (irnnr) [inline Classes/ConnectionManager.php]: You could also assign the connection to a variable then pass it to you function and eventually push it onto `$connections`

File: Classes/SolrService.php [hunk 2]
```diff
@@ -610,8 +610,9 @@ public function getSolrconfigName()
             $solrconfigXmlUrl = $this->_scheme . '://'
                 . $this->_host . ':' . $this->_port
                 . $this->_path . 'admin/file/?file=solrconfig.xml';
+            $response= $this->_sendRawGet($solrconfigXmlUrl);
```
Comment (irnnr) [inline Classes/SolrService.php]: Is this supposed to be part of this PR?

File: Documentation/Configuration/Reference/TxSolrSolr.rst [hunk 3]
```diff
@@ -69,3 +69,23 @@ path
 
 Sets the path portion of the URL. Make sure to have the path end with a slash (/).
 
+username
+--------
+
+:Type: String
+:TS Path: plugin.tx_solr.solr.username
+:Default: /
```
Comment (irnnr) [inline Documentation/Configuration/Reference/TxSolrSolr.rst]: If there's no default value just remove this line

**Comment by irnnr** [comment]
Looks good, just some minor changes and questions.

**Comment by timohund** [comment]
@pgampe  Thanks for your effort and for preparing a pull request.

I would agree to @irnnr  comments. In additionn i have some other remarks:

* The next version is 6.0, this is wrong in the documentation
* I would propose to avoid end($array) and instead spitup the call into asignment and then call the methode and finally add it to the array, or instead have $solrUser and $solrPassword as additional optional arguments of getConnection()

File: Documentation/Configuration/Reference/TxSolrSolr.rst (hunk 3 above)
Comment (pgampe) [inline Documentation/Configuration/Reference/TxSolrSolr.rst]: oops, will do

**Comment by pgampe** [comment]
Will look into the comments.

File: Classes/SolrService.php (hunk 2 above)
Comment (irnnr) [inline Classes/SolrService.php]: ok

**Comment by irnnr** [comment]
as @timohund said, if you add the parameters to the constructor that's fine, too.

**Comment by timohund** [comment]
@pgampe Thanks for your adaptions. The branch has merge conflicts with the master. Can you please rebase your branch?

**Comment by pgampe** [comment]
Sorry,  missed the default stuff from Ingo.

**Comment by timohund** [comment]
@pgampe Looks good to me, did you also check what happens when no authentication is configured.

@irnnr can you approve this change now?

**Comment by pgampe** [comment]
The last time I checked, it worked fine without authentication, but I did not retest the latest rebases and master stuff. But from reading, it should only kick in if the username is set to an non-empty string.

**Pull request merged**